### PR TITLE
Add checks for Data.Traversable.Compat

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

**Comment by sol** [comment]
@RyanGlScott again, review would be much appreciated.

File: mk-index.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+runhaskell dumpindex.hs Prelude.Compat          | sort > check/Prelude.Compat.index
+runhaskell dumpindex.hs Data.Foldable.Compat    | sort > check/Data.Foldable.Compat.index
+runhaskell dumpindex.hs Data.List.Compat        | sort > check/Data.List.Compat.index
```
Comment (RyanGlScott) [inline mk-index.sh]: Should `Data.Traversable.Compat.index` be added to this file?

File: run-check.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
-runhaskell dumptypes.hs Prelude | typediff check/Prelude.types -
-runhaskell dumptypes.hs Data.Foldable | typediff check/Data.Foldable.types -
-runhaskell dumptypes.hs Data.List | typediff check/Data.List.types -
+runhaskell dumptypes.hs Prelude.Compat | typediff check/Prelude.Compat.types -
+runhaskell dumptypes.hs Data.Foldable.Compat | typediff check/Data.Foldable.Compat.types -
+runhaskell dumptypes.hs Data.List.Compat | typediff check/Data.List.Compat.types -
```
Comment (RyanGlScott) [inline run-check.sh]: Similarly, should a `Data.Traversable.Compat.types` file be created and checked here?

File: mk-index.sh (hunk 1 above)
Comment (sol) [inline mk-index.sh]: Haha, yeah, good catch.  Shouldn't hack when I'm tried...

**Comment by sol** [comment]
Updated.

**Comment by RyanGlScott** [comment]
Awesome!

Now we just need to generalise (and check) `Control.Monad.Compat`, and that should take care of all the craziness brought about by the FTP/AMP.

**Comment by sol** [comment]
Yes.

BTW, note that result from `dumpindex.hs` still needs some manual editing of the output for some modules.

**Pull request merged**